### PR TITLE
lesspipe: fix macOS build

### PR DESCRIPTION
**Pull request by muneebmahmed**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

The recent 2.07 release results in this error on macOS:
```
/usr/local/bin/lesspipe.sh: line 398: typeset: -l: invalid option
typeset: usage: typeset [-afFirtx] [-p] name[=value] ...
```
likely due to [this line](https://github.com/wofr06/lesspipe/blob/a24c496dd1d820795be8ddc624d65d8504fb5dab/lesspipe.sh#L398). This seems to be a problem with macOS `bash` being outdated.
Adding a dependency on Homebrew's `bash` seems unnecessary, as this formula is [compatible with zsh](https://github.com/wofr06/lesspipe/blob/a24c496dd1d820795be8ddc624d65d8504fb5dab/configure#L88), so we can just use `/bin/zsh` for macOS builds instead.

### CODE REVIEW

File: Formula/lesspipe.rb [hunk 1]
```diff
@@ -16,7 +17,12 @@ class Lesspipe < Formula
   end
 
   def install
-    system "./configure", "--prefix=#{prefix}"
+    args = %W[
+      --prefix=#{prefix}
+    ]
+    # Newer shell features not present in macOS bash
```
Comment (chenrui333) [inline Formula/lesspipe.rb:23]: should we just `depends_on bash` for the new shell features?

File: Formula/lesspipe.rb [hunk 2]
```diff
@@ -16,7 +17,12 @@ class Lesspipe < Formula
   end
 
   def install
-    system "./configure", "--prefix=#{prefix}"
+    args = %W[
+      --prefix=#{prefix}
+    ]
```
Comment (SMillerDev) [inline Formula/lesspipe.rb:22]:
```suggestion
    args = *std_configure_args
```

File: Formula/lesspipe.rb (hunk 1 above)
Comment (muneebmahmed) [reply Formula/lesspipe.rb:23]: This alone would be insufficient. Their configure script [strips out](https://github.com/wofr06/lesspipe/blob/a24c496dd1d820795be8ddc624d65d8504fb5dab/configure#L122) the dirname of the shell and uses only `#!/usr/bin/env $selected_shell` so homebrew bash would also need to be earlier in the PATH.
Some options I can think of are:
* Alter the configure script upstream to not do this
* Substitute this line with a direct link to homebrew's bash
* Submit a PR upstream to not use the instance of `typeset -l`
* Use pre-installed zsh

My preference was for switching to zsh, as this doesn't install a new dependency just for a script

File: Formula/lesspipe.rb (hunk 2 above)
Comment (muneebmahmed) [reply Formula/lesspipe.rb:22]: This gives an error as the the script exits on the unknown arguments `disable-debug`, `disable-dependency-tracking` and `libdir`. Is there another way to give just the single `prefix` argument?

Comment (SMillerDev) [reply Formula/lesspipe.rb:22]: No, if it's not a real configure script this is fine.

File: Formula/lesspipe.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/lesspipe.rb:23]: > Submit a PR upstream to not use the instance of typeset -l

can we log an upstream issue for that, I feel like that is a better approach, no?

Comment (muneebmahmed) [reply Formula/lesspipe.rb:23]: Opened an issue for it here: https://github.com/wofr06/lesspipe/issues/121

**Comment by muneebmahmed** [comment]
I updated this PR to instead use the upstream patch. However, I'm unable to test it so I don't know why it's failing:

```
HOMEBREW_INSTALL_FROM_API=0 brew test lesspipe
Error: this command cannot be run while HOMEBREW_INSTALL_FROM_API is set!
```

Can you point me to instructions to unset this variable? It's not in any of my shell configs and I didn't find instructions to unset it in the official documentation

**Comment by SMillerDev** [comment]
Could you rebase this pull request?

**Comment by muneebmahmed** [comment]
> Could you rebase this pull request?

Done

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**